packages/core/src/providers/tiktok.js

```diff
@@ -0,0 +1,36 @@
+/** @type {import(".").OAuthProvider} */
+export default function TikTok(options) {
+  return {
+    id: "tiktok",
+    name: "TikTok",
+    type: "oauth",
+    version: "2.0",
+    authorization: {
+      url: "https://www.tiktok.com/auth/authorize/",
+      params: {
+        scope: "user.info.basic",
+        response_type: "code",
+        client_key: options.clientId,
+        redirect_uri: options.redirect_uri,
+      },
+    },
+    token: {
+      url: "https://open-api.tiktok.com/oauth/access_token/",
+      params: {
+        client_key: options.clientId,
+        client_secret: options.clientSecret,
+        grant_type: "authorization_code",
+      },
+    },
+    userinfo: "https://open-api.tiktok.com/user/info/",
+    profile(profile) {
+      return {
+        id: profile.open_id,
+        name: profile.display_name,
+        image: profile.avatar_url
```

Comment:
```suggestion
        image: profile.avatar_url,
        email: profile.email ?? null
```

Do you know if e-mail can be returned?

Reply:
https://developers.tiktok.com/doc/tiktok-api-v2-get-user-info/  if you scroll down on this page, there is docs for the User Object - I couldn't find any ref to get an email unfortunately